src_Core/CPU/CPU.bsv

```diff
@@ -866,7 +984,21 @@ module mkCPU (CPU_IFC);
 		     rg_next_seq <= stage1.out.data_to_stage2.instr_seq + 1;
 `endif
 		     redirect = True;
+`ifdef PERFORMANCE_MONITORING
+		     events.evt_REDIRECT = True;
+`endif
 		  end
+`ifdef PERFORMANCE_MONITORING
+		  if (   (stage1.out.data_to_stage2.op_stage2 == OP_Stage2_ST)
+		      && (stage1.out.data_to_stage2.mem_width_code == w_SIZE_CAP)   ) begin
+		     events.evt_MEM_CAP_STORE = True;
+		     CapReg capReg = cast (extract_cap (stage1.out.data_to_stage2.val2));
+		     CapMem capMem = cast (capReg);
+		     events.evt_MEM_CAP_STORE_TAG_SET = isValidCap (capMem);
+		  end
+		  events.evt_IMPRECISE_SETBOUND =  ! stage1.out.data_to_stage2.check_exact_success;
```

Comment:
Most of the time this is `?`; it's only valid when `check_exact_enable` is `True`.